src/api/query-hooks/index.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from "@tanstack/react-query";
+import { getIncident } from "../services/incident";
+import { getTopologyComponents } from "../services/topology";
+
+const defaultStaleTime = 1000 * 60 * 5;
+
+export const createIncidentQueryKey = (id: string) => ["getIncident", id];
+
+export const useIncidentQuery = (id = "") => {
+  return useQuery(createIncidentQueryKey(id), () =>
+    getIncident(id).then((response) => response.data)
+  );
+};
+
+export const useComponentsQuery = () => {
+  return useQuery(["allcomponents"], getTopologyComponents, {
```

Comment:
I don't think we want to use a topology cache as the only source for id lookups, we should have a seperate cache of just id to -> name, icon, type with very long expiration e.g. 2h 

the logic should be:

check local name cache, 
       check full topology cache (with 5m expiry)
            get /component/<id>